### PR TITLE
#409 - docker.buildArg.myBuildArg to docker build-arg both from proje…

### PR DESCRIPTION
**Pull request by balazsmaria**
…ct and system properties

----------------------
branch [[build-args-from-outside]]

### CODE REVIEW

File: src/main/java/io/fabric8/maven/docker/AbstractBuildSupportMojo.java [hunk 1]
```diff
@@ -63,7 +68,27 @@ protected void buildImage(ServiceHub hub, ImageConfiguration imageConfig)
...
+        return ImmutableMap.<String, String>builder().putAll(buildArgsFromProject).putAll(buildArgsFromSystem).build();
+    }
+
+    private Map<String, String> addBuildArgsFromProperties(Properties properties) {
+        String argPrefix  = "docker.buildArg.";
+        Map<String, String> buildArgs = new HashMap<>();
+        for(Object keyObj : properties.keySet()){
+            String key = (String)keyObj;
+            if (key.startsWith(argPrefix)){
+                String argKey = key.replace(argPrefix, "");
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/AbstractBuildSupportMojo.java]: would use either `replaceFirst` or `substring(argPrefix.length()` for the unlikely case that someone use a strange name for its build arg :)

**Comment by rhuss** [comment]
:+1:  Fancy to update the documentation, too ? (in docker-build.md)

thanks ! :)

File: src/main/java/io/fabric8/maven/docker/AbstractBuildSupportMojo.java (hunk 1 above)
Comment (balazsmaria) [inline src/main/java/io/fabric8/maven/docker/AbstractBuildSupportMojo.java]: You're right here.

**Comment by balazsmaria** [comment]
I added a Build-args part. Hope that's fine.

**Comment by rhuss** [comment]
LGTM, thanks a lot !

(will rearrange the docs a bit, thats a minor thinggy)

**Pull request merged**